pytorch_lightning/accelerators/ddp_slurm_backend.py

```diff
@@ -0,0 +1,179 @@
+# limitations under the License
+import os
+import torch
+import torch.distributed as torch_distrib
+import torch.distributed as dist
+
+from pytorch_lightning.accelerators.base_backend import Accelerator
+from pytorch_lightning import _logger as log
+from pytorch_lightning.utilities import AMPType
+from pytorch_lightning.utilities.distributed import rank_zero_only
+from pytorch_lightning.utilities.seed import seed_everything
+from pytorch_lightning.distributed.dist import LightningDistributed
+
+
+try:
+    from hydra.utils import to_absolute_path, get_original_cwd
+    from hydra.core.hydra_config import HydraConfig
+except ImportError:
+    HYDRA_AVAILABLE = False
+else:
+    HYDRA_AVAILABLE = True
+
+
+# -------------------------------------------
+# !!!!!!!!!!!!!! NOTE !!!!!!!!!!!!!!!!!!!!!!
+# TEMP CLASS WHILE WE DECOUPLE SLURM FROM DDP
+# !!!!!!!!!!!!!! NOTE !!!!!!!!!!!!!!!!!!!!!!
+# -------------------------------------------
+class DDPSLURMBackend(Accelerator):
+
+    def __init__(self, trainer):
+        super().__init__(trainer)
+        self.task_idx = None
+        self._has_spawned_children = False
+        self.dist = LightningDistributed()
+
+    def setup(self, model):
+        self.trainer.model = model
+        self.task_idx = int(os.environ['SLURM_LOCALID'])
+
+    def train(self):
+        model = self.trainer.model
+        self.ddp_train(process_idx=self.task_idx, model=model)
+
+    def set_world_ranks(self, process_idx):
+        self.trainer.local_rank = process_idx
+        self.trainer.global_rank = self.trainer.node_rank * self.trainer.num_processes + process_idx
+        self.trainer.world_size = self.trainer.num_nodes * self.trainer.num_processes
+
+    def model_to_device(self, model, process_idx):
+        self.trainer.root_gpu = process_idx
+        torch.cuda.set_device(self.trainer.root_gpu)
+        model.cuda(self.trainer.root_gpu)
+
+    def get_device_ids(self):
+        device_ids = [self.trainer.root_gpu]
+        return device_ids
+
+    def training_step(self, args):
+        if self.trainer.amp_backend == AMPType.NATIVE:
+            with torch.cuda.amp.autocast():
+                output = self.trainer.model(*args)
+        else:
+            output = self.trainer.model(*args)
+        return output
+
+    def validation_step(self, args):
+        output = self.training_step(args)
+        return output
+
+    def test_step(self, args):
+        output = self.training_step(args)
+        return output
+
+    def barrier(self, name: str = None):
+        if torch_distrib.is_initialized():
+            torch_distrib.barrier()
+
+    def early_stopping_should_stop(self, pl_module):
+        stop = torch.tensor(int(self.trainer.should_stop), device=pl_module.device)
+        dist.all_reduce(stop, op=dist.reduce_op.SUM)
+        dist.barrier()
+        should_stop = stop == self.trainer.world_size
+        return should_stop
+
+    def broadcast(self, obj, src=0):
+        return self.dist.broadcast(obj)
```

Comment:
why's src unused?

Reply:
this is an override of the parent method in Accelerator. It's to keep behavior consistent
across accelerators.

src is already set into this object automatically.